sympy/matrices/expressions/matpow.py

```diff
@@ -1,24 +1,24 @@
 from __future__ import print_function, division
 
 from sympy.matrices.common import NonSquareMatrixError
-from .matexpr import MatrixExpr, Identity, ZeroMatrix
-from sympy.core import S
+from .matexpr import MatrixExpr, Identity
+from sympy.core import S, Basic
 from sympy.core.sympify import _sympify
 from sympy.matrices import MatrixBase
-from sympy.matrices.common import NonInvertibleMatrixError
-
-from .permutation import PermutationMatrix
 
 
 class MatPow(MatrixExpr):
 
     def __new__(cls, base, exp, evaluate=False, **options):
         base = _sympify(base)
         if not base.is_Matrix:
-            raise TypeError("Function parameter should be a matrix")
+            raise TypeError("MatPow base should be a matrix")
+        if not base.is_square:
+            raise NonSquareMatrixError("Power of non-square matrix %s" % base)
+
         exp = _sympify(exp)
+        obj = Basic.__new__(cls, base, exp)
 
-        obj = super(MatPow, cls).__new__(cls, base, exp)
```

Comment:
Why is this skipping the superclass?

That seems to be common in matrix expressions code but I don't think it is good practice.

Reply:
I'm actually not sure, I just saw that it's being done like this in many places. I can restore it.